### PR TITLE
Add ability to pull SNAPSHOT releases from snapshots.elastic.co

### PR DESCRIPTION
**Pull request by robbavey**
Elasticsearch 7.x requires Java 11 to build, which Logstash is currently
incompatible with. This commit pulls SNAPSHOT releases from the
snapshots site, skipping the need to build Elasticsearch and allowing
integration tests against 7.x to run

### CODE REVIEW

**Comment by robbavey** [comment]
@jsvd Wasn't quite ready ;) 

I've updated the PR to enable tests to be run using the latest snapshot for a version - specify `LATEST-SNAPSHOT-x`, and the build script will interrogate the endpoint at `https://artifacts-api.elastic.co/v1/versions/` for the latest version matching version `x`. So `LATEST-SNAPSHOT-6` would pull the last snapshot release of 6 (currently `6.6.0-SNAPSHOT`), and `LATEST-SNAPSHOT-6.4` would pull the last snapshot release of 6.4 (currently `6.4.4-SNAPSHOT`)

**Comment by jsvd** [comment]
Haha you gotta mark these as WIP :D
I just saw an improvement and lgtmd :P

**Comment by robbavey** [comment]
@jsvd Are you ok with the extra changes?

File: .travis.yml [hunk 1]
```diff
@@ -6,10 +6,11 @@ env:
...
-- DISTRIBUTION=legacy INTEGRATION=true ES_VERSION=5.6.9 TEST_DEBUG=true
-- DISTRIBUTION=oss INTEGRATION=true ES_VERSION=6.4.2 TEST_DEBUG=true
-- DISTRIBUTION=default INTEGRATION=true ES_VERSION=6.4.2 TEST_DEBUG=true
-- DISTRIBUTION=default INTEGRATION=true ES_VERSION=master TEST_DEBUG=true
+- DISTRIBUTION=legacy INTEGRATION=true ES_VERSION=5.6.13 TEST_DEBUG=true
+- DISTRIBUTION=oss INTEGRATION=true ES_VERSION=6.4.3 TEST_DEBUG=true
+- DISTRIBUTION=default INTEGRATION=true ES_VERSION=6.4.3 TEST_DEBUG=true
+- DISTRIBUTION=default INTEGRATION=true ES_VERSION=LATEST-SNAPSHOT-6 TEST_DEBUG=true
+- DISTRIBUTION=default INTEGRATION=true ES_VERSION=LATEST-SNAPSHOT-7 TEST_DEBUG=true
 - DISTRIBUTION=legacyxpack SECURE_INTEGRATION=true INTEGRATION=true ES_VERSION=5.6.9 TEST_DEBUG=true
```
Comment (jsvd) [inline .travis.yml]: while we're here maybe we can also update this 5.6.9 and the 6.4.2 below?

**Pull request closed without merging**